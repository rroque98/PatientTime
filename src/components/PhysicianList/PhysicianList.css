.notableHeader {
  color: var(--notableBlue);
  font-size: 2em;
  font-weight: 600;
}
.physListHeader {
  font-size: 1em;
  font-weight: 600;
  padding: 1em 0 0 0;
}

li {
  list-style-type: circle;
}

ul li {
  color: var(--darkGrey);
}

ul li span {
  color: var(--black);
}

button {
  display: block;
  margin: auto;
  padding: 1em;
  background: var(--notableBlue);
  color: var(--white);
  border-radius: 6px;
  text-align: center;
  outline: none;
  font-size: 1em;
}

button:hover {
  background: var(--darkBlue);
}
