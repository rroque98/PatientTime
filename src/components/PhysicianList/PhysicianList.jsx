import React from "react";

const PhysicianList = ({ physicians, updateApptsOnClick }) => {
  return (
    <div>
      <h1>Notable</h1>
      <h3>Physicians</h3>
      <ul>
        {physicians.map(physician => (
          <li
            key={physician.id}
            onClick={e => updateApptsOnClick(e, physician.id)}
          >
            {physician.lastName}, {physician.firstName}
          </li>
        ))}
      </ul>
      <button>Logout</button>
    </div>
  );
};

export default PhysicianList;
