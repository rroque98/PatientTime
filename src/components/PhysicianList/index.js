export { default } from "./PhysicianList.jsx";
