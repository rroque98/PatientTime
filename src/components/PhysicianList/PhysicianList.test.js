import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
import PhysicianList from "./index.js";
import "jest-styled-components";

afterEach(cleanup);
const physicians = [
  {
    id: 1,
    firstName: "Morris",
    lastName: "Coppedge"
  },
  {
    id: 2,
    firstName: "Evela",
    lastName: "Lonergan"
  },
  {
    id: 3,
    firstName: "Random",
    lastName: "Name"
  },
  {
    id: 4,
    firstName: "Elizabeth",
    lastName: "Walker"
  },
  {
    id: 5,
    firstName: "Rani",
    lastName: "Peltarik"
  }
];
test("should find Logout button", () => {
  const { getByText } = render(
    <PhysicianList physicians={physicians} currPhysId={1} />
  );
  const elem = getByText("Logout");
  expect(elem.innerHTML).toBe("Logout");
  fireEvent.click(elem);
});

// test("Does the Logout button have the correct styles", () => {
//   const { getByText } = render(
//     <PhysicianList physicians={physicians} currPhysId={1} />
//   );
//   const elem = getByText("Logout");
//   expect(elem).toHaveSyleRule("display", "block");
//   expect(elem).toHaveSyleRule("font-size", "1em");
//   expect(elem).toHaveSyleRule("font-weight", "600");
// });

test("PhysicianList snapshot", () => {
  const physListComp = render(
    <PhysicianList physicians={physicians} currPhysId={1} />
  );
  expect(physListComp).toMatchSnapshot();
});
