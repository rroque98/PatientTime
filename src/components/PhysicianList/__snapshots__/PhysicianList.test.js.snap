// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhysicianList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  list-style-type: circle;
  line-height: 2em;
  list-style-position: inside;
  text-indent: 1em;
}

.c0 {
  font-size: 2em;
  font-weight: 600;
  padding: 0.5em;
}

.c1 {
  font-size: 1em;
  font-weight: 600;
  padding: 1em 1em 0 1em;
}

.c2 {
  padding: 0;
}

.c4 {
  display: block;
  margin: 0 auto 1em auto;
  padding: 0.5em 1em;
  border-radius: 6px;
  text-align: center;
  outline: none;
  font-size: 1em;
  font-weight: 600;
}

<body>
    <div>
      <header
        class="c0"
      >
        PatientTime
      </header>
      <header
        class="c1"
      >
        PHYSICIANS
      </header>
      <ul
        class="c2"
      >
        <li
          class="select c3"
        >
          <span
            class=""
          >
            Coppedge
            , 
            Morris
          </span>
        </li>
        <li
          class="c3"
        >
          <span
            class=""
          >
            Lonergan
            , 
            Evela
          </span>
        </li>
        <li
          class="c3"
        >
          <span
            class=""
          >
            Name
            , 
            Random
          </span>
        </li>
        <li
          class="c3"
        >
          <span
            class=""
          >
            Walker
            , 
            Elizabeth
          </span>
        </li>
        <li
          class="c3"
        >
          <span
            class=""
          >
            Peltarik
            , 
            Rani
          </span>
        </li>
      </ul>
      <button
        class="c4"
      >
        Logout
      </button>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-htpNat ifndpt"
    >
      PatientTime
    </header>
    <header
      class="sc-bxivhb kViTCW"
    >
      PHYSICIANS
    </header>
    <ul
      class="sc-ifAKCX JnUSj"
    >
      <li
        class="select sc-bdVaJa gVmRUz"
      >
        <span
          class="sc-bwzfXH jivAqU"
        >
          Coppedge
          , 
          Morris
        </span>
      </li>
      <li
        class="sc-bdVaJa gVmRUz"
      >
        <span
          class="sc-bwzfXH jivAqU"
        >
          Lonergan
          , 
          Evela
        </span>
      </li>
      <li
        class="sc-bdVaJa gVmRUz"
      >
        <span
          class="sc-bwzfXH jivAqU"
        >
          Name
          , 
          Random
        </span>
      </li>
      <li
        class="sc-bdVaJa gVmRUz"
      >
        <span
          class="sc-bwzfXH jivAqU"
        >
          Walker
          , 
          Elizabeth
        </span>
      </li>
      <li
        class="sc-bdVaJa gVmRUz"
      >
        <span
          class="sc-bwzfXH jivAqU"
        >
          Peltarik
          , 
          Rani
        </span>
      </li>
    </ul>
    <button
      class="sc-EHOje jfcqvd"
    >
      Logout
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
